### ISSUE TITLE
"Reached a code path in Module.get_extra_state() that should never be called."

### ISSUE THREAD

**Issue by albertsgarde** [pytorch/pytorch#90373]
### 🐛 Describe the bug

I got the run time error mentioned in the title when calling `get_extra_state` on a `RecursiveScriptModule` created from an `nn.Dropout` module.
I was just playing around so don't know if it even makes sense, but since the error specifically told me to report it I thought I should.
If nothing else, this error should probably be changed.

Image of entire error stack: 
![image](https://user-images.githubusercontent.com/20074572/206148420-032e8035-2921-4839-bb86-ee58b1419c73.png)

### Versions

Collecting environment information...
PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: (tdm64-1) 9.2.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.6 | packaged by conda-forge | (main, Oct 24 2022, 16:02:16) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19044-SP0
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2060
Nvidia driver version: 516.94
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] torch==1.13.0
[pip3] torchaudio==0.13.0
[pip3] torchvision==0.14.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2021.4.0           haa95532_640
[conda] mkl-service               2.4.0           py310h2bbff1b_0
[conda] mkl_fft                   1.3.1           py310ha0764ea_0
[conda] mkl_random                1.2.2           py310h4ed8f06_0
[conda] numpy                     1.23.3          py310h60c9a35_0
[conda] numpy-base                1.23.3          py310h04254f7_0
[conda] pytorch                   1.13.0          py3.10_cuda11.7_cudnn8_0    pytorch
[conda] pytorch-cuda              11.7                 h67b0de4_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.13.0                   pypi_0    pypi
[conda] torchvision               0.14.0                   pypi_0    pypi

**Comment by jingxu10** [comment]
Hi, would it be possible to share a test case for reproduction? Thank you.
